test/integration/redis_proxy_integration_test.cc

```diff
@@ -0,0 +1,83 @@
+#include "test/integration/redis_proxy_integration_test.h"
+
+#include <sstream>
+#include <vector>
+
+#include "extensions/filters/network/redis_proxy/command_splitter_impl.h"
+
+namespace RedisCommon = Envoy::Extensions::NetworkFilters::Common::Redis;
+namespace RedisCmdSplitter = Envoy::Extensions::NetworkFilters::RedisProxy::CommandSplitter;
+
+namespace Envoy {
+namespace {
+
+std::string makeBulkStringArray(std::vector<std::string>&& command_strings) {
+  std::stringstream result;
+
+  result << "*" << command_strings.size() << "\r\n";
+  for (uint64_t i = 0; i < command_strings.size(); i++) {
+    result << "$" << command_strings[i].size() << "\r\n";
+    result << command_strings[i] << "\r\n";
+  }
+
+  return result.str();
+}
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, RedisProxyIntegrationTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+void RedisProxyIntegrationTest::initialize() {
+  config_helper_.renameListener("redis_proxy");
+  BaseIntegrationTest::initialize();
+}
+
+TEST_P(RedisProxyIntegrationTest, SimpleRequestAndResponse) {
+  initialize();
+  IntegrationTcpClientPtr tcp_client = makeTcpConnection(lookupPort("redis_proxy"));
+
+  std::string client_to_proxy = makeBulkStringArray({"get", "foo"});
+  std::string proxy_to_server;
+
+  ASSERT_TRUE(client_to_proxy.size() > 0);
```

Comment:
nit: in general prefer EXPECT vs. ASSERT unless there is a good reason to use ASSERT.

Reply:
done